scripts/validate-metadata.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# PACKAGE METADATA
+# ----------------
+if [[ $(python setup.py --name) == "UNKNOWN" ]] ; then
+    echo "[error] metadata: name missing"
+    exit 2
+fi
+
+if [[ $(python setup.py --version) == "0.0.0" ]] ; then
+    echo "[error] metadata: version missing"
+    exit 2
+fi
+
+# TODO: This presently breaks due to setuptools normalizing the package version to something that may not conform to semantic versioning in some cases.
+#       On future versions of setuptools this issue should be solved (a patch was jsut merged) so we can get back to this.
+#       See: https://github.com/pypa/setuptools/issues/308
+if ! python setup.py --version | grep -q -E '^((([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$' ; then
```

Comment:
If Marco approves the use of `Linux/Python Compatible Semantic Versioning 3.0.0` [See here](https://docs.openstack.org/pbr/latest/user/semver.html) then we can remove the check as the generated version will be PEP 440 compliant.

Reply:
I see. The 3.0.0 would allow us to constraint it to the git tag as well, since it would accommodate fort those cases that do not conform to the old versions. The regex would need to change though.
If you have a link to the new regex please let me know or add it to the code.

Reply:
I might miss a case, but this seems to work OK

`^([0-9]+)\.([0-9]+)\.([0-9]+)([-\.\+]?[0-9a-zA-Z]+)?([-\.\+]?[0-9a-zA-Z]+)*$`

Please check it out at: [regexr](https://regexr.com/4b171)

Reply:
This is addressed at https://github.com/ska-telescope/lmc-base-classes/pull/97